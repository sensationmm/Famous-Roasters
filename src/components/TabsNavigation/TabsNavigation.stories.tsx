import { ComponentMeta, ComponentStory } from '@storybook/react'
import React from 'react'

import { TabsNavigation } from './TabsNavigation'

export default {
  title: 'Components/TabsNavigation',
  component: TabsNavigation,
} as ComponentMeta<typeof TabsNavigation>

const Template: ComponentStory<typeof TabsNavigation> = (args) => <TabsNavigation {...args} />

export const Default = Template.bind({})
Default.args = {
  tabsData: [
    { key: 'forYou', translationKey: 'pages.catalogue.tabs.forYou' },
    { key: 'discover', translationKey: 'pages.catalogue.tabs.discover' },
  ],
  initialActiveTabKey: 'discover',
  setParentActiveTab: () => null,
}
