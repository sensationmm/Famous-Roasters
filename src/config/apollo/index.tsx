import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
