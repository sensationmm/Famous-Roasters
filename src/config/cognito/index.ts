import useAuth from './auth.hook'

export { useAuth as useAuth }
