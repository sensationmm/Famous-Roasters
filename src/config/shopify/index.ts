export const shopifyCoffeeCollection = 'gid://shopify/Collection/386655387864'
export const shopifyAccessoryCollection = 'gid://shopify/Collection/385139605720'
