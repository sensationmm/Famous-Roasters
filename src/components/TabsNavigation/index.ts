import { TabsNavigation } from './TabsNavigation'

export { TabsNavigation as TabsNavigation }
