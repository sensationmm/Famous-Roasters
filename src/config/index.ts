import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
