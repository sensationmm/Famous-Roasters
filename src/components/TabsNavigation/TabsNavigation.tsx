import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Typography, TypographySize } from 'src/components'

export interface TabsDataItem {
  key: string
  translationKey: string
}

interface TabsNavigationProps {
  tabsData: TabsDataItem[]
  initialActiveTabKey?: string
  setParentActiveTab: (key: string) => void
}

export const TabsNavigation: React.FC<TabsNavigationProps> = ({
  tabsData,
  initialActiveTabKey = tabsData[0].key,
  setParentActiveTab,
}: TabsNavigationProps) => {
  const { t } = useTranslation()
  const [activeTab, setActiveTab] = useState<string>(initialActiveTabKey)

  const handleTabItemClick = (key: string) => {
    setActiveTab(key)
    setParentActiveTab(key)
  }

  return (
    <div className="flex" role="tablist">
      {tabsData.map((tabsDataItem: TabsDataItem) => {
        const isActive = tabsDataItem.key === activeTab
        return (
          <button
            type="button"
            onClick={() => handleTabItemClick(tabsDataItem.key)}
            key={tabsDataItem.key}
            data-testid={`tab-${tabsDataItem.key}`}
            className="mr-6"
            role="tab"
            aria-selected={isActive}
          >
            <Typography
              size={TypographySize.Small}
              className={
                isActive
                  ? 'text-brand-black font-semibold border-b-2 pb-1.5'
                  : 'text-coreUI-text-secondary hover:text-coreUI-text-primary pb-1.5'
              }
            >
              {t(tabsDataItem.translationKey)}
            </Typography>
          </button>
        )
      })}
    </div>
  )
}
