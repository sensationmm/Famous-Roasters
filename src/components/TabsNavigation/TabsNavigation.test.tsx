import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'

import { i18n } from '../../config'
import { TabsNavigation } from './TabsNavigation'

global.alert = jest.fn()

const tabsData = [
  { key: 'forYou', translationKey: 'pages.catalogue.tabs.forYou' },
  { key: 'discover', translationKey: 'pages.catalogue.tabs.discover' },
]

describe('Tabs Navigation component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TabsNavigation tabsData={tabsData} setParentActiveTab={(key) => alert(key)} />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can navigate across tabs', async () => {
    render(
      <I18nextProvider i18n={i18n}>
        <TabsNavigation tabsData={tabsData} setParentActiveTab={(key) => alert(key)} />
      </I18nextProvider>,
    )
    const tabForYou = await screen.findByTestId('tab-forYou')
    const tabDiscover = await screen.findByTestId('tab-discover')
    expect(tabForYou).toBeInTheDocument()
    expect(tabDiscover).toBeInTheDocument()
    fireEvent.click(tabForYou)
    fireEvent.click(tabDiscover)
  })
})
