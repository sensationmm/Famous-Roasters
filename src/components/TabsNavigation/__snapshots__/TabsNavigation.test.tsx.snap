// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Navigation component Renders correctly 1`] = `
<div>
  <div
    class="flex"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="mr-6"
      data-testid="tab-forYou"
      role="tab"
      type="button"
    >
      <span
        class="text-brand-black font-semibold border-b-2 pb-1.5 text-sm leading-5"
      >
        FÃ¼r Dich
      </span>
    </button>
    <button
      aria-selected="false"
      class="mr-6"
      data-testid="tab-discover"
      role="tab"
      type="button"
    >
      <span
        class="text-coreUI-text-secondary pb-1.5 text-sm leading-5"
      >
        Entdecken
      </span>
    </button>
  </div>
</div>
`;
